File: cash_c/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float input = 0;
    int coins = 0;

    do
    {
        input = get_float("Input your change owed: ");
    }
    while (input <= 0);

    int cents = round(input * 100);

    for (int i = cents; i >= 0; i--)
    {
        if (cents >= 25)
        {
            cents -= 25;
            coins += 1;
        }
        else if (cents >= 10)
        {
            cents -= 10;
            coins += 1;
        }
        else if (cents >= 5)
        {
            cents -= 5;
            coins += 1;
        }
        else if (cents >= 1)
        {
            cents -= 1;
            coins += 1;
        }
    }

    printf("coins: %i\n", coins);

}

File: cash_py/cash.py
from cs50 import get_float

# ask for user input until gets the positive integer
while True:
    cash = get_float("Input your change owed: ")
    if cash >= 0:
        break

# converting cash to cents
cents = cash*100
# minimum number of coins for make change
coins = 0

while cents != 0:
    if cents >= 25:
        cents -= 25
        coins += 1
    elif cents >= 10:
        cents -= 10
        coins += 1
    elif cents >= 5:
        cents -= 5
        coins += 1
    else:
        cents -= 1
        coins += 1

print(coins)